File: src/main/java/ee/leola/kassa/models/Product.java
package ee.leola.kassa.models;

import jakarta.persistence.Entity;
import jakarta.validation.constraints.NotNull;
import java.math.BigDecimal;
import lombok.Data;
import lombok.EqualsAndHashCode;

@Entity
@Data
@EqualsAndHashCode(callSuper = true)
public class Product extends Model {

  @NotNull private String name;

  @NotNull private BigDecimal price = BigDecimal.ZERO;

  private Integer quantity;

  public void incrementQuantity(Integer increment) {
    setQuantity(getQuantity() + increment);
  }
}

File: src/main/kotlin/ee/leola/kassa/models/Product.kt
package ee.leola.kassa.models

import jakarta.persistence.Entity
import jakarta.validation.constraints.NotNull
import java.math.BigDecimal

@Entity
class Product(
    var name: @NotNull String,
    var price: @NotNull BigDecimal = BigDecimal.ZERO,
    var quantity: Int,
) : Model() {
    fun incrementQuantity(increment: Int): Product {
        quantity += increment
        return this
    }
}
